pkg/operator/controller/ingress/status.go

```diff
@@ -619,13 +619,27 @@ func computeLoadBalancerStatus(ic *operatorv1.IngressController, service *corev1
 		// Try and find a more specific reason for for the pending status.
 		createFailedReason := "SyncLoadBalancerFailed"
 		failedLoadBalancerEvents := getEventsByReason(operandEvents, "service-controller", createFailedReason)
-		for _, event := range failedLoadBalancerEvents {
-			involved := event.InvolvedObject
-			if involved.Kind == "Service" && involved.Namespace == service.Namespace && involved.Name == service.Name && involved.UID == service.UID {
-				reason = "SyncLoadBalancerFailed"
-				message = fmt.Sprintf("The %s component is reporting SyncLoadBalancerFailed events like: %s\n%s",
-					event.Source.Component, event.Message, "The kube-controller-manager logs may contain more details.")
-				break
+		if len(failedLoadBalancerEvents) > 0 {
+			for _, event := range failedLoadBalancerEvents {
+				involved := event.InvolvedObject
+				if involved.Kind == "Service" && involved.Namespace == service.Namespace && involved.Name == service.Name && involved.UID == service.UID {
+					reason = "SyncLoadBalancerFailed"
+					message = fmt.Sprintf("The %s component is reporting SyncLoadBalancerFailed events like: %s\n%s",
+						event.Source.Component, event.Message, "The kube-controller-manager logs may contain more details.")
+					break
+				}
+			}
+		} else {
+			// The LB service eventually stops reporting events, even if they
+			// are still valid. If the service is still in the pending state
+			// and on the previous update, it was in a failed state, assume it
+			// is still in a failed state even if no events are present.
+			for _, condition := range ic.Status.Conditions {
+				if condition.Type == operatorv1.LoadBalancerReadyIngressConditionType &&
+					condition.Reason == "SyncLoadBalancerFailed" {
+					conditions = append(conditions, condition)
```

Comment:
Do you think you should add a log here, because the BZ was inquiring about a lack of log messages?

Reply:
The log the BZ is asking about is the installer log, which I don't think we have a direct way to post messages to, but the degraded status condition already includes the error from `LoadBalancerReady` if there is one, and I believe the degraded message does get included in the installer log if ingress stays degraded.